platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserService.java

```diff
@@ -280,12 +280,12 @@ public UserDescriptor getById(@ApiParam(value = "User ID") @PathParam("id") Stri
     @GenerateLink(rel = LINK_REL_GET_USER_BY_EMAIL)
     @RolesAllowed({"user", "system/admin", "system/manager"})
     @Produces(APPLICATION_JSON)
-    public UserDescriptor getByEmail(@ApiParam(value = "User email", required = true) @QueryParam("email") @Required String email,
+    public UserDescriptor getByAlias(@ApiParam(value = "User alias", required = true) @QueryParam("alias") @Required String alias,
                                      @Context SecurityContext context) throws NotFoundException, ServerException, ConflictException {
-        if (email == null) {
-            throw new ConflictException("Missed parameter email");
+        if (alias == null) {
+            throw new ConflictException("Missed parameter alias");
```

Comment:
I understand that all you did is variable rename, but anyway don't you think that changing of this code to throw `BadRequestException`  would improve this service, or maybe you can change all the methods of the service to throw `BadRequestExcetpion` in the case of not valid data.